cloudinit/net/network_manager.py

```diff
@@ -116,6 +170,14 @@ def _set_ip_method(self, family, subnet_type):
         self.config[family]["method"] = method
         self._set_default(family, "may-fail", "false")
 
+        # we do not want to set may-fail to false for both ipv4 and ipv6 dhcp
+        # at the at the same time. This will make the network configuration
+        # work only when both ipv4 and ipv6 dhcp succeeds. This may not be
+        # what we want. If we have configured both ipv4 and ipv6 dhcp, any one
+        # succeeding should be enough. Therefore, if "may-fail" is set to
+        # False for both ipv4 and ipv6 dhcp, set them both to True.
+        self._set_mayfail_true_if_both_false_dhcp()
```

Comment:
I don't think this is the right spot for this as we'll run it for every subnet.

I included a patch to this review that moves it to after we configure the subnets.

Reply:
Yes good idea.